src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReferenceDataController.kt

```diff
@@ -189,4 +194,15 @@ class ReferenceDataController(
 
     return ResponseEntity.ok(referralRejectionReasons.map(referralRejectionReasonTransformer::transformJpaToApi))
   }
+
+  override fun referenceDataPrisonReleaseTypesGet(
+    xServiceName: ServiceName?,
+  ): ResponseEntity<List<PrisonReleaseType>> {
+    val prisonReleaseTypes = when (xServiceName != null) {
+      true -> prisonReleaseTypeRepository.findAllByServiceScope(xServiceName.value)
+      else -> prisonReleaseTypeRepository.findAll(Sort.by(Sort.Direction.ASC, "sortOrder"))
```

Comment:
I think we should deliberately only support CAS3 if we're not sure if this endpoint will be used by other CASs in the future e.g.

```
    if (xServiceName != ServiceName.temporaryAccommodation) {
      throw ForbiddenProblem()
    }
```

If we do this CAS3 can happily change this endpoint without having to worry about whether CAS1/CAS2 is using it, which can sometimes be hard to determine without looking at UI code. I guess we can also remove the default findAll function too if we do this?